trin-core/src/jsonrpc/types.rs

```diff
@@ -34,6 +38,27 @@ pub struct JsonRequest {
     pub id: u32,
 }
 
+// Global portal network JSON-RPC request
+#[derive(Debug, Clone)]
+pub struct PortalJsonRpcRequest {
+    pub endpoint: PortalEndpointKind,
+    pub resp: Responder<Value, String>,
+}
+
+/// History network JSON-RPC request
+#[derive(Debug)]
+pub struct HistoryJsonRpcRequest {
+    pub endpoint: HistoryEndpointKind,
+    pub resp: Responder<Value, String>,
+}
+
+/// State network JSON-RPC request
+#[derive(Debug)]
+pub struct StateJsonRpcRequest {
```

Comment:
Personally, I think these distinct types are fine for now. @jacobkaufmann mentioned some possible consolidation - which I also think is a good idea - similarly I'm sure we could consolidate the `StateEndpointKind::DataRadius`/ `HistoryEndpointKind::DataRadius` in some way - but I really think these are ok, and maybe even preferable in terms of readability.